guibot/finder.py

```diff
@@ -104,6 +104,26 @@ def __repr__(self):
         return ("<value='%s' min='%s' max='%s' delta='%s' tolerance='%s' fixed='%s' enumerated='%s'>"
                 % (self.value, self.min_val, self.max_val, self.delta, self.tolerance, self.fixed, self.enumerated))
 
+    def __eq__(self, other):
+        """
+        Custom implementation for equality check.
+
+        :returns: whether this instance is equal to another
+        :rtype: bool
+        """
+        if not isinstance(other, CVParameter):
+            return NotImplemented
```

Comment:
This is a new approach towards a lack of implementation as we usually `raise NotImplementedError`. Does it have some advantages? Also, perhaps it is better to throw `ValueError` or some other error that signifies comparing incompatible types and we are not planning to implement this in the future (right?).

Reply:
It's the [recommended approach](https://docs.python.org/3/reference/datamodel.html#object.__eq__):
> A rich comparison method may return the singleton NotImplemented if it does not implement the operation for a given pair of arguments. By convention, False and True are returned for a successful comparison.